extension/storage/filestorage/client.go

```diff
@@ -117,40 +142,77 @@ func (c *fileStorageClient) Close(_ context.Context) error {
 }
 
 // Compact database. Use temporary file as helper as we cannot replace database in-place
-func (c *fileStorageClient) Compact(ctx context.Context, compactionDirectory string, timeout time.Duration, maxTransactionSize int64) (*fileStorageClient, error) {
+func (c *fileStorageClient) Compact(ctx context.Context, compactionDirectory string, timeout time.Duration, maxTransactionSize int64) error {
+	var err error
+	var file *os.File
+	var compactedDb *bbolt.DB
+	compactionStart := time.Now()
```

Comment:
I think we should separately time the blocking part of this function - the one after locking `compactionMutex`. That is the most practically relevant metric here, as we don't much care how much time it takes to prepare the new db.

Reply:
Yeah, I think only the latter actually makes sense